src/diag.c

```diff
@@ -260,3 +260,97 @@ void uvc_print_diag(uvc_device_handle_t *devh, FILE *stream) {
   }
 }
 
+
+/** @brief Print all possible frame configuration.
+ * @ingroup diag
+ *
+ * @param devh UVC device
+ * @param stream Output stream (stderr if NULL)
+ */
+void uvc_print_frameformats(uvc_device_handle_t *devh) {
+
+  if (devh->info->ctrl_if.bcdUVC) {
+    uvc_streaming_interface_t *stream_if;
+    int stream_idx = 0;
+    DL_FOREACH(devh->info->stream_ifs, stream_if) {
+      uvc_format_desc_t *fmt_desc;
+      ++stream_idx;
+
+      DL_FOREACH(stream_if->format_descs, fmt_desc) {
+        uvc_frame_desc_t *frame_desc;
+        int i;
+
+        switch (fmt_desc->bDescriptorSubtype) {
+          case UVC_VS_FORMAT_UNCOMPRESSED:
+          case UVC_VS_FORMAT_MJPEG:
+          case UVC_VS_FORMAT_FRAME_BASED:
+            printf("         \%s(%d)\n"
```

Comment:
hmh, not sure, same as line 184 ... formatting looks right in output in both instances